sycl/plugins/unified_runtime/ur/adapters/hip/queue.cpp

```diff
@@ -0,0 +1,280 @@
+    return UR_RESULT_SUCCESS;
+  }
+
+  try {
+    std::unique_ptr<ur_queue_handle_t_> QueueImpl(hQueue);
+
+    ScopedContext Active(hQueue->getContext());
+
+    hQueue->forEachStream([](hipStream_t S) {
+      UR_CHECK_ERROR(hipStreamSynchronize(S));
+      UR_CHECK_ERROR(hipStreamDestroy(S));
+    });
+
+    return UR_RESULT_SUCCESS;
+  } catch (ur_result_t Err) {
+    return Err;
+  } catch (...) {
+    return UR_RESULT_ERROR_OUT_OF_RESOURCES;
+  }
+}
+
+UR_APIEXPORT ur_result_t UR_APICALL urQueueFinish(ur_queue_handle_t hQueue) {
+  // set default result to a negative result (avoid false-positve tests)
+  ur_result_t Result = UR_RESULT_ERROR_OUT_OF_RESOURCES;
+
+  try {
+
+    ScopedContext Active(hQueue->getContext());
+
+    hQueue->syncStreams<true>([&Result](hipStream_t S) {
+      Result = UR_CHECK_ERROR(hipStreamSynchronize(S));
+    });
+
+  } catch (ur_result_t Err) {
+    Result = Err;
+  } catch (...) {
+    Result = UR_RESULT_ERROR_OUT_OF_RESOURCES;
+  }
+
+  return Result;
+}
+
+// There is no HIP counterpart for queue flushing and we don't run into the
+// same problem of having to flush cross-queue dependencies as some of the
+// other plugins, so it can be left as no-op.
+UR_APIEXPORT ur_result_t UR_APICALL urQueueFlush(ur_queue_handle_t hQueue) {
+  std::ignore = hQueue;
+  return UR_RESULT_SUCCESS;
+}
+
+/// Gets the native HIP handle of a UR queue object
+///
+/// \param[in] hQueue The UR queue to get the native HIP object of.
+/// \param[out] phNativeQueue Set to the native handle of the UR queue object.
+///
+/// \return UR_RESULT_SUCCESS
+UR_APIEXPORT ur_result_t UR_APICALL
+urQueueGetNativeHandle(ur_queue_handle_t hQueue, ur_queue_native_desc_t *pDesc,
+                       ur_native_handle_t *phNativeQueue) {
+  std::ignore = pDesc;
+
+  ScopedContext Active(hQueue->getContext());
+  *phNativeQueue =
+      reinterpret_cast<ur_native_handle_t>(hQueue->getNextComputeStream());
+  return UR_RESULT_SUCCESS;
+}
+
+/// Created a UR queue object from a HIP queue handle.
+/// TODO: Implement this.
+/// NOTE: The created UR object takes ownership of the native handle.
+///
+/// \param[in] hNativeQueue The native handle to create UR queue object from.
+/// \param[in] hContext is the UR context of the queue.
+/// \param[out] phQueue Set to the UR queue object created from native handle.
+/// \param pProperties->isNativeHandleOwned tells if SYCL RT should assume the
+/// ownership of
+///        the native handle, if it can.
+///
+/// \return UR_RESULT_ERROR_UNSUPPORTED_FEATURE
+UR_APIEXPORT ur_result_t UR_APICALL urQueueCreateWithNativeHandle(
+    ur_native_handle_t hNativeQueue, ur_context_handle_t hContext,
+    ur_device_handle_t hDevice, const ur_queue_native_properties_t *pProperties,
+    ur_queue_handle_t *phQueue) {
+
+  std::ignore = hNativeQueue;
+  std::ignore = hContext;
+  std::ignore = hDevice;
+  std::ignore = pProperties;
+  std::ignore = phQueue;
+  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
```

Comment:
It seems like doing this is becoming the standard way of dealing with this https://github.com/intel/llvm/pull/8303/commits/f2a2026c16de51b68de60b503414f93aa43fb699#diff-8da59f0730e70207732121ceff2865ddb20a577d248e7cf4709658de4fc8a579R19 (I copied from other PRs)
Currently I don't think the programmer will get any error message currently otherwise.

Reply:
We were trying to avoid calling `die` as much as possible as that would terminate the program, I would suggest that we could resolve the `die` call in another PR as that might prove controversial and is connected to the way that adapters specific error handling would be done. Also, I would prefer that the port here not change any behaviour that was there as that could introduce other unexpected problems.